core/templates/services/UpgradedServices.ts

```diff
@@ -798,6 +797,8 @@ export class UpgradedServices {
         upgradedServices['WrittenTranslationObjectFactory']);
 
     // Topological level: 2.
+    upgradedServices['CsrfTokenService'] = new CsrfTokenService(
+      upgradedServices['HttpXhrBackend']);
```

Comment:
Why `HttpXhrBackend` is used instead of `HttpBackend`?

Reply:
I use it because HttpClient also uses it 

```typescript
upgradedServices['HttpClient'] = new HttpClient(
      upgradedServices['HttpXhrBackend']);
```